src/Core/Client/Adapter/TimeoutAwareTrait.php

```diff
@@ -8,9 +8,9 @@
 trait TimeoutAwareTrait
 {
     /**
-     * @var int|null
+     * @var int
      */
-    private $timeout;
+    private $timeout = TimeoutAwareInterface::DEFAULT_TIMEOUT;
 
     public function setTimeout(int $timeoutInSeconds): void
```

Comment:
do we actually need this to be nullable? I think currently on the endpoint the timeout can be disabled (set to null)? or not?